sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java

```diff
@@ -0,0 +1,571 @@
+		if(mSdlEncoder != null) {
+			mSdlEncoder.drainEncoder(endOfStream);
+		}
+	}
+
+	/**
+	 * Opens a video service (service type 11) and subsequently provides an IVideoStreamListener
+	 * to the app to send video data. The supplied VideoStreamingParameters will be set as desired paramaters
+	 * that will be used to negotiate
+	 *
+	 * @param parameters  Video streaming parameters including: codec which will be used for streaming (currently, only
+	 *                    VideoStreamingCodec.H264 is accepted), height and width of the video in pixels.
+	 * @param encrypted Specify true if packets on this service have to be encrypted
+	 *
+	 * @return IVideoStreamListener interface if service is opened successfully and streaming is
+	 *         started, null otherwise
+	 */
+	public IVideoStreamListener startVideoStream(VideoStreamingParameters parameters, boolean encrypted){
+		if(hmiLevel != HMILevel.HMI_FULL){
+			Log.e(TAG, "Cannot start video service if HMILevel is not FULL.");
+			return null;
+		}
+		IVideoStreamListener listener = internalInterface.startVideoStream(encrypted, parameters);
+		if(listener != null){
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		}else{
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+		}
+		return listener;
+	}
+
+	/**
+	 * Starts video service, sets up encoder, haptic manager, and remote display. Begins streaming the remote display.
+	 * @param parameters Video streaming parameters including: codec which will be used for streaming (currently, only
+	 *                    VideoStreamingCodec.H264 is accepted), height and width of the video in pixels.
+	 * @param encrypted Specify true if packets on this service have to be encrypted
+	 */
+	private void startVideoStreaming(VideoStreamingParameters parameters, boolean encrypted){
+		IVideoStreamListener streamListener = startVideoStream(parameters, encrypted);
+		if(streamListener == null){
+			Log.e(TAG, "Error starting video service");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			return;
+		}
+		VideoStreamingCapability capability = (VideoStreamingCapability) internalInterface.getCapability(SystemCapabilityType.VIDEO_STREAMING);
+		if(capability != null && capability.getIsHapticSpatialDataSupported()){
+			hapticManager = new HapticInterfaceManager(internalInterface);
+		}
+		try {
+			encoder.init(this.context.get(),streamListener,parameters);
+			//We are all set so we can start streaming at at this point
+			encoder.start();
+			//Encoder should be up and running
+			createRemoteDisplay(encoder.getVirtualDisplay());
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		} catch (Exception e) {
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			e.printStackTrace();
+		}
+		Log.d(TAG, parameters.toString());
+	}
+
+	/**
+	 * Stops streaming service and remote display encoder if applicable.
+	 */
+	public void stopStreaming(){
```

Comment:
Is there a reason this is `public`?

Reply:
To give a way to stop streaming, I suppose `dispose` does the same thing, so this could be `private`

Reply:
Ok, no I think that is smart. If stopped is called, the manager has to be able to restart properly without being recreated and as long as that is possible a `stopStreaming` method would be useful for developers. I would add javadoc to this method that explains what steps (at a high level) would need to be done to restart the stream.